driver/tests/main/test_restart_serial.py

```diff
@@ -0,0 +1,113 @@
+        hours=0,
+        minutes=0,
+        seconds=225,
+        layout=(1, 1),
+        initialization=unittest.mock.MagicMock(),
+        performance_config=unittest.mock.MagicMock(),
+        comm_config=NullCommConfig((1, 1)),
+        diagnostics_config=unittest.mock.MagicMock(),
+        dycore_config=unittest.mock.MagicMock(nwat=6),
+        physics_config=unittest.mock.MagicMock(),
+    )
+    assert driver_config.save_restart is False
+
+
+def test_restart_save_to_disk():
+    try:
+        backend = "gtc:numpy"
+        mpi_comm = NullComm(rank=0, total_ranks=6, fill_value=0.0)
+        partitioner = pace.util.CubedSpherePartitioner(
+            pace.util.TilePartitioner((1, 1))
+        )
+        communicator = pace.util.CubedSphereCommunicator(mpi_comm, partitioner)
+        sizer = pace.util.SubtileGridSizer.from_tile_params(
+            nx_tile=12,
+            ny_tile=12,
+            nz=79,
+            n_halo=3,
+            extra_dim_lengths={},
+            layout=(1, 1),
+            tile_partitioner=partitioner.tile,
+            tile_rank=communicator.tile.rank,
+        )
+        quantity_factory = pace.util.QuantityFactory.from_backend(
+            sizer=sizer, backend=backend
+        )
+
+        init = BaroclinicConfig()
+        driver_state = init.get_driver_state(
+            quantity_factory=quantity_factory, communicator=communicator
+        )
+        driver_state.save_state(mpi_comm)
+
+        restart_dycore = xr.open_dataset(
+            f"RESTART/restart_dycore_state_{mpi_comm.rank}.nc"
+        )
+        for var in driver_state.dycore_state.__dict__.keys():
+            if isinstance(driver_state.dycore_state.__dict__[var], pace.util.Quantity):
+                np.testing.assert_allclose(
+                    driver_state.dycore_state.__dict__[var].data,
+                    restart_dycore[var].values,
+                )
+            else:
+                if var in restart_dycore.keys():
+                    raise KeyError(
+                        f"{var} is not a quantity and \
+                        should not be in dycore restart file"
+                    )
+
+        restart_physics = xr.open_dataset(
+            f"RESTART/restart_physics_state_{mpi_comm.rank}.nc"
+        )
+        for var in driver_state.physics_state.__dict__.keys():
+            if isinstance(
+                driver_state.physics_state.__dict__[var],
+                gt4py.storage.storage.CPUStorage,
+            ):
+                np.testing.assert_allclose(
+                    driver_state.physics_state.__dict__[var].data,
+                    restart_physics[var].values,
+                )
+            else:
+                if var in restart_physics.keys():
+                    raise KeyError(
+                        f"{var} is not a storage and \
+                            should not be in physics restart file"
+                    )
+    finally:
+        subprocess.Popen(["make", "clean"])
```

Comment:
Please change this to something like `shutil.rmdir("RESTART")`. In the future if we can make the restart output path configurable, we could tell it to write to a tmpdir.